libs/core/src/lib/micro-process-flow/components/micro-process-flow/micro-process-flow.component.ts

```diff
@@ -0,0 +1,379 @@
+        this.listenOnItemsChange();
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._subscriptions.unsubscribe();
+    }
+
+    /** @hidden */
+    @HostListener('keydown', ['$event'])
+    handleKeyboardEvent(event: KeyboardEvent): void {
+        if (KeyUtil.isKeyCode(event, this._navigationKeys)) {
+            const isRightKey = KeyUtil.isKeyCode(event, RIGHT_ARROW);
+            const direction = isRightKey ? 1 : -1;
+            event.stopImmediatePropagation();
+
+            const elementIndexToScroll = this._getPreviousItemsCount()  + direction;
+            const elementExists = this.items.get(elementIndexToScroll);
+
+            if (!elementExists) {
+                return;
+            }
+
+            this.previousItemsCount = 0;
+
+            this._paginate(elementIndexToScroll);
+
+            // Force browset not to scroll to the element since it's done with pagination function.
+            this.items.get(this._getPreviousItemsCount())?.focusableElement?.focus({
+                preventScroll: true
+            });
+        }
+    }
+
+    /** Listens on items change and checks if navigation buttons should be visible. */
+    listenOnItemsChange(): void {
+        this._subscriptions.add(this.items.changes.pipe(startWith(0)).subscribe(() => {
+            this.items.forEach((item) => {
+                item.setLastItem(false);
+            });
+
+            this.items.last?.setLastItem(true);
+            this._setNavigationButtons();
+            this._paginate(0);
+        }));
+    }
+
+    /** Scrolls to the next item. */
+    goNext(): void {
+        this._paginate();
+    }
+
+    /** Scrolls to the previous item. */
+    goBack(): void {
+        this._paginate(-1);
+    }
+
+    /**
+     * Checks if navigation buttons should be visible depending on the amount of items
+     * and current pagination offset.
+     */
+     private _setNavigationButtons(): void {
+        if (this._container === undefined) {
+            return;
+        }
+
+        const containerWidth = this._container.nativeElement.offsetWidth;
+        const itemsWidth = this._wrapperContainer.nativeElement.offsetWidth;
+        const nextButtonWidth = this._goNextButton?.nativeElement.offsetWidth || 0;
+
+        if (itemsWidth < containerWidth) {
+            return;
+        }
+
+        // Skip previously shown items out of the calculation.
+        const itemsToDisplay = this.items
+            .toArray()
+            .slice(this.previousItemsCount)
+            .map((i) => i.elRef);
+
+        // Check if all items can be fitted inside the container if we remove 'next' button.
+        if (this.previousItemsCount > 0) {
+            const remainingWidth = itemsToDisplay.reduce((width, elm) => elm.nativeElement.offsetWidth + width, 0);
+
+            if (remainingWidth <= nextButtonWidth + containerWidth) {
+                this.nextItemsCount = 0;
+                this._cd.detectChanges();
```

Comment:
can we use `markForCheck` here?

Reply:
No, it fails to render pagination appropriately